src/lib.rs

```diff
@@ -187,6 +191,28 @@ fn find_node_by_weight<Ty: EdgeType>(
     Ok(index)
 }
 
+fn merge_duplicates<K, V, F, E>(
+    xs: Vec<(K, V)>,
+    mut merge_fn: F,
+) -> Result<Vec<(K, V)>, E>
+where
+    K: Hash + Eq,
+    F: FnMut(&V, &V) -> Result<V, E>,
+{
+    let mut kvs = DictMap::with_capacity(xs.len());
```

Comment:
Is there a reason to use a `DictMap` here as opposed to using just a `HashMap`, like do we need to preserve order? Normally `HashMap` is a bit faster.

Reply:
I think it's not so much that we need to preserve order here, but more that we want a deterministic order based on the input `Vec` contents. This is because for our use cases, we take the result of this function and add edges to the graph following its iteration order. It seems like `DictMap` is an easy way to get this.